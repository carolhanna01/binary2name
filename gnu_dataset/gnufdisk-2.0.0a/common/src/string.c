/* GNU Fidsk (gnufdisk-common), a library of common functions.
 *
 * Copyright (C) 2011 Free Software Foundation
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA. */

#include <string.h>
#include <stdlib.h>
#include <stdarg.h>
#include <errno.h>

#include <gnufdisk-common.h>

struct gnufdisk_string {
  char* data;
  size_t size;
};

static int check_string(struct gnufdisk_string* _s)
{
  if(gnufdisk_check_memory(_s, sizeof(struct gnufdisk_string), 0) != 0
     || (_s->data != NULL && gnufdisk_check_memory(_s->data, _s->size, 0) != 0))
    return EFAULT;
  return 0;
}

struct gnufdisk_string* gnufdisk_string_new(const char* _fmt, ...)
{
  va_list args;
  struct gnufdisk_string* ret;
  int err;

  if((ret = malloc(sizeof(struct gnufdisk_string))) == NULL)
    {
      err = errno;
      goto lb_out;
    }

  memset(ret, 0, sizeof(struct gnufdisk_string));
    
  va_start(args, _fmt);
  ret->size = gnufdisk_vasprintf(&ret->data, _fmt, args);
  va_end(args);

  if(ret->size < 0)
    {
      err = errno;
      
      free(ret);
      ret = NULL;
    }

lb_out:
  
  errno = err;
  return ret;
}

int gnufdisk_string_set(struct gnufdisk_string* _p, const char* _fmt, ...)
{
  va_list args;
  char* s;
  int err;
  int ret;

  if((err = check_string(_p)) != 0)
    {
      ret = -1;
      goto lb_out;
    }
  
  va_start(args, _fmt);
  ret = gnufdisk_vasprintf(&s, _fmt, args);
  va_end(args);

  if(ret < 0)
    {
      ret = -1;
      err = errno;
      goto lb_out;
    }
  else 
    err = 0;

  if(_p->data)
    free(_p->data);

  _p->data = s;
  _p->size = ret;

lb_out:
  
  errno = err;
  return ret;
}

int gnufdisk_string_length(struct gnufdisk_string* _p)
{
  if((errno = check_string(_p)) != 0)
    return -1;

  return strlen(_p->data != NULL ? _p->data : "");
}

const char* gnufdisk_string_c_string(struct gnufdisk_string* _p)
{
  if((errno = check_string(_p)) != 0)
    return NULL;

  return _p->data;
}

char* gnufdisk_string_c_string_dup(struct gnufdisk_string* _p)
{
  char *r;

  if((errno = check_string(_p)) != 0)
    return NULL;

  r = strdup(_p->data ? _p->data : "");

  if(r == NULL)
    errno = ENOMEM;

  return r;
}

int gnufdisk_string_delete(struct gnufdisk_string* _s)
{
  if((errno = check_string(_s)) != 0)
    return -1;

  if(_s->data)
    free(_s->data);

  free(_s);

  return 0;
}

