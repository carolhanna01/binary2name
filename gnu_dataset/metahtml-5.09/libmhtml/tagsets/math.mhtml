;;; math.mhtml: -*- Meta-HTML -*-  A few math helpers.
;;; 
;;;  Copyright (c) 1998 Brian J. Fox
;;;  Author: Brian J. Fox (bfox@ai.mit.edu) Sun Apr 19 13:21:39 1998.
<defsubst float whitespace=delete>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns the floating point representation of <var x>.
  ;;; <var x> may be an integer, a floating point number, or a variable
  ;;; containing an integer or floating point number.
  ;;; <complete-example>
  ;;; <set-var foo=7>
  ;;; <float foo>
  ;;; <float 3>
  ;;; <float 3.0>
  ;;; </complete-example>
  <mul 1.0 %0>
</defsubst>

<defun number? arg &key base>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns "true" if <var arg> is the string representation of an integer
  ;;; in base <var base> (default <code>10</code>), or the string
  ;;; representation of a floating point number in base 10.
  ;;;
  ;;; For integer checks, the special value of zero (<code>0</code>) for
  ;;; <var base> allows the common radixes of decimal, octal, and hexadecimal
  ;;; to be understood.  That is to say:
  ;;; <example>
  ;;; <number? <get-var x> base=0>
  ;;; </example>
  ;;; is equivalent to:
  ;;; <example>
  ;;; <or <integer? <get-var x> base=8>
  ;;;     <integer? <get-var x> base=10>
  ;;;     <integer? <get-var x> base=16>
  ;;;     <real?    <get-var x>>>
  ;;; </example>
  ;;;
  ;;; Some examples:
  ;;; <example>
  ;;; <number? 10>          --> true
  ;;; <number? .9>          --> true
  ;;; <number 0xEF base=16> --> true
  ;;; </example>
  <defvar base 10>
  <or <integer? <get-var arg> base=<get-var base>> <real? <get-var arg>>>
</defun>

<defun avg &rest nums[]>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns the average of all of the arguments passed.
  ;;; Examples:
  ;;; <example>
  ;;; <avg 3 4 5>    --> 4
  ;;; <avg 4 9 3.2>  --> 5.40
  ;;; </example>
  <set-var divisor = <array-size nums>>
  <set-var dividend = <%%backquote-eval <add ,<get-var nums[]>>>>
  <div dividend divisor>
</defun>

<defun factorial x>
  ;;; DOC_SECTIONS (ARITHMETIC-OPERATORS)
  ;;; Computes the factorial of <var x>.
  ;;; Example:
  ;;; <complete-example>
  ;;; <factorial 12>
  ;;; </complete-example>
  <if <lt x 2>
      1
    <mul <float x> <fact <sub x 1>>>>
</defun>
