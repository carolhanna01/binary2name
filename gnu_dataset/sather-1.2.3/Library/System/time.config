----------------->  GNU Sather - compiler configuration file  <----------------
-- Copyright (C) 199x by International Computer Science Institute            --
-- This file is part of the GNU Sather library. It is free software; you may --
-- redistribute  and/or modify it under the terms of the GNU Library General --
-- Public  License (LGPL)  as published  by the  Free  Software  Foundation; --
-- either version 3 of the license, or (at your option) any later version.   --
-- This  library  is distributed  in the  hope that it will  be  useful, but --
-- WITHOUT ANY WARRANTY without even the implied warranty of MERCHANTABILITY --
-- or FITNESS FOR A PARTICULAR PURPOSE. See Doc/LGPL for more details.       --
-- The license text is also available from:  Free Software Foundation, Inc., --
-- 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA                     --
-------------->  Please email comments to <bug-sather@gnu.org>  <--------------

#ifndef TIME_CONFIG
#define TIME_CONFIG

#ifdef PP
#undef PP
#endif

#ifdef APP
#undef APP
#endif

#ifdef VPP
#undef VPP
#endif

#ifdef FPP
#undef FPP
#endif

#ifdef VSTD
#undef VSTD
#endif

#define PP	no_pre,no_post
#define VPP	volatile,PP
#define VSTD	var "@r $$;"

TIME_SECONDS:	VPP,VSTD,exec "$$=(FLTD)time(NULL);" "$$";

TIMES_NOW:	VPP,declare "#include <sys/times.h>" "#include <limits.h>",
	        var "struct tms $$;",
	        exec "$0->$.wall_time=times(&$$);"
		     "$0->$.sys_time=$$.tms_stime;"
		     "$0->$.user_time=$$.tms_utime;"
		     "$0->$.child_sys_time=$$.tms_cstime;"
		     "$0->$.child_user_time=$$.tms_cutime;";
TIMES_CLK:	PP,VSTD,exec "$$=CLK_TCK;",declare "#include <limits.h>";

MEMORY_NOW:	PP,exec "$0->$.sbrk_size=(long)sbrk(0);"
		        "$0->$.gc_size=(long)GC_get_heap_size();";

#endif 
