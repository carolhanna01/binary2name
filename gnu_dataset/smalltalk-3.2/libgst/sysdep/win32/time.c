/******************************** -*- C -*- ****************************
 *
 * System specific implementation module.
 *
 * This module contains implementations of various operating system
 * specific routines.  This module should encapsulate most (or all)
 * of these calls so that the rest of the code is portable.
 *
 *
 ***********************************************************************/

/***********************************************************************
 *
 * Copyright 1988,89,90,91,92,94,95,99,2000,2001,2002,2003,2006,2007,2008,2009
 * Free Software Foundation, Inc.
 * Written by Steve Byrne.
 *
 * This file is part of GNU Smalltalk.
 *
 * GNU Smalltalk is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation; either version 2, or (at your option) any later
 * version.
 *
 * Linking GNU Smalltalk statically or dynamically with other modules is
 * making a combined work based on GNU Smalltalk.  Thus, the terms and
 * conditions of the GNU General Public License cover the whole
 * combination.
 *
 * In addition, as a special exception, the Free Software Foundation
 * give you permission to combine GNU Smalltalk with free software
 * programs or libraries that are released under the GNU LGPL and with
 * independent programs running under the GNU Smalltalk virtual machine.
 *
 * You may copy and distribute such a system following the terms of the
 * GNU GPL for GNU Smalltalk and the licenses of the other code
 * concerned, provided that you include the source code of that other
 * code when and as the GNU GPL requires distribution of source code.
 *
 * Note that people who make modified versions of GNU Smalltalk are not
 * obligated to grant this special exception for their modified
 * versions; it is their choice whether to do so.  The GNU General
 * Public License gives permission to release a modified version without
 * this exception; this exception also makes it possible to release a
 * modified version which carries forward this exception.
 *
 * GNU Smalltalk is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * GNU Smalltalk; see the file COPYING.	 If not, write to the Free Software
 * Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 *
 ***********************************************************************/


#include "gstpriv.h"

#ifdef HAVE_SYS_TIMES_H
# include <sys/times.h>
#endif

#ifdef HAVE_SYS_TIMEB_H
#include <sys/timeb.h>
#endif

#ifdef WIN32
# define WIN32_LEAN_AND_MEAN /* avoid including junk */
# include <windows.h>
#endif

uint64_t
_gst_get_milli_time (void)
{
  static long frequency = 0, frequencyH;
  long milli;
  LARGE_INTEGER counter;

  if (!frequency)
    {
      QueryPerformanceFrequency (&counter);
      /* frequencyH = 1000 * 2^32 / frequency */
      frequency = counter.LowPart;
      frequencyH = MulDiv (1000 * (1 << 16), (1 << 16), counter.LowPart);
    }

  QueryPerformanceCounter (&counter);
  /* milli = (high * 2^32 + low) * 1000 / freq =
     high * (1000 * 2^32 / freq) + (low * 1000 / freq) =
     (high * frequencyH) + (low / 4) * 4000 / freq)
     
     Dividing and multiplying counter.LowPart by 4 is needed because
     MulDiv accepts signed integers but counter.LowPart is unsigned.  */
  milli = counter.HighPart * frequencyH;
  milli += MulDiv (counter.LowPart >> 2, 4000, frequency);
  return milli;
}

char *
_gst_current_time_zone_name (void)
{
  char *zone;
  long bias = _gst_current_time_zone_bias () / 60;
  TIME_ZONE_INFORMATION tzi;
  LPCWSTR name;
  static char buffer[32];
  GetTimeZoneInformation (&tzi);
  zone = buffer;
  name = (bias == (tzi.Bias + tzi.StandardBias))
  	  ? tzi.StandardName : tzi.DaylightName;

  WideCharToMultiByte (CP_ACP, 0, name, lstrlenW (name), zone, 32, NULL, NULL);
  return xstrdup (zone);
}

void
_gst_usleep (int us)
{
  Sleep (us / 1000);
}
