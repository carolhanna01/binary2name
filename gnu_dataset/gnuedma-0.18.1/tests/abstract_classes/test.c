/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <signal.h>
#include <time.h>
#include <unistd.h>

#include <edma.h>

ESint32 EDMAPROC my_met1 (OBJID, EPChar, ESint32);
ESint32 EDMAPROC my_met2 (OBJID, EPChar);

int
main(int argc,char *argv[])
{
  OBJID    id;

  /* Initialize EDMA System */
  EDMAInit ();
  
  /* Optionally you can create a file named edma_repos.lst with
   * the list of repositories you want to load at application 
   * start up
   */
  edma_add_std_repo ("edma_test.cfg");
  printf ("%s", "----------------------------------\n");

  /* Create an object of class ABSTRACT2
   * ABSTRACT2 is a class containing some abstract methods
   * met1 is pure abstract
   */
  if ((id = edma_new_obj ("ABSTRACT2")) == -1) 
    {
      printf ("%s", "Can't create ABSTRACT2 object. Aborting");
      EDMAEnd ();
      exit (1);
    }

  /* GNU/EDMA default abstract method handler will be executed for met1
   * a warnning will be shown in console the other methods are executed
   * normally.
   */
  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");
  edma_met3 (id, "met3", "Hello");
  printf ("%s", "----------------------------------\n");
  
  /* Override virtual methods */
  /* Virtual methods on abstract class ABSTRACT2 are overriden
   * by main application functions 
   */
  edma_over_met (id, "met1", NULL, (PPROC) my_met1);
  edma_over_met (id, "met2", NULL, (PPROC) my_met2);

  /* Re-run ABSTRACT2 method. met3 is still bind to default abstract method */
  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");
  edma_met3 (id, "met3", "Hello");
  printf ("%s", "----------------------------------\n");

  /* Finish with current object */
  edma_free_obj (id);

  /* Now, we test, pure abstract clases*/
  if ((id = edma_new_obj ("ABSTRACT1")) == -1) 
    {
      printf ("%s", "Can't create ABSTRACT1 object");
      EDMAEnd ();
      exit (1);
    }

  /* ABSTRACT1 is a pute virtual class. No method implementation */
  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");  

  /* Override virtual methods on ABSTRACT1 */
  edma_over_met (id, "met1", NULL, (PPROC) my_met1);
  edma_over_met (id, "met2", NULL, (PPROC) my_met2);
  printf ("%s", "----------------------------------\n");
  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");
  edma_free_obj (id);
  printf ("%s", "----------------------------------\n");

  /* Now, we test a realization of this interface*/
  if ((id = edma_new_obj ("REALIZATION")) == -1) 
    {
      printf ("%s", "Can't create ABSTRACT1 object");
      EDMAEnd ();
      exit (1);
    }

  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");  

  /* Override virtual methods*/
  edma_over_met (id, "met1", NULL, (PPROC) my_met1);
  edma_over_met (id, "met2", NULL, (PPROC) my_met2);
  printf ("%s", "----------------------------------\n");
  edma_met3 (id, "met1", "Hello", 2);
  edma_met3 (id, "met2", "Hello");
  edma_free_obj (id);

  /* Shutdown EDMA Suystem */
  EDMAEnd ();
  return 0;
}

/* Overrider functions*/
ESint32 EDMAPROC 
my_met1 (OBJID IdObj, EPChar a, ESint32 b) 
{
  edma_printf_obj (IdObj, "New code for met1. Par1: '%s' Par2: %d",a ,b);
  
  return 0;
}

ESint32 EDMAPROC 
my_met2 (OBJID IdObj, EPChar a) 
{
  edma_printf_obj (IdObj, "New code for met2. Par1: '%s'",a);
  
  return 0;
}
