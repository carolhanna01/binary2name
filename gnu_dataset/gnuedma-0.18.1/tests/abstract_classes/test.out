GNU/EDMA 0.16.1 build [Jan 23 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
----------------------------------
(ABSTRACT2)[SYSTEM_WARNNING] Trying to execute abstract method 'met1' on instance 0 of class 'ABSTRACT2'
(ABSTRACT2)This is a virtual method with default implementation
(ABSTRACT2)This is a normal methods
----------------------------------
(ABSTRACT2)New code for met1. Par1: 'Hello' Par2: 2
(ABSTRACT2)New code for met2. Par1: 'Hello'
(ABSTRACT2)This is a normal methods
----------------------------------
(ABSTRACT1)[SYSTEM_WARNNING] Trying to execute abstract method 'met1' on instance 0 of class 'ABSTRACT1'
(ABSTRACT1)[SYSTEM_WARNNING] Trying to execute abstract method 'met2' on instance 0 of class 'ABSTRACT1'
----------------------------------
(ABSTRACT1)New code for met1. Par1: 'Hello' Par2: 2
(ABSTRACT1)New code for met2. Par1: 'Hello'
----------------------------------
(REALIZATION)[met1] String: Hello | Integer : 2
(REALIZATION)[met2] String : Hello
**ERROR**[edma_over_met] Method met1 in Class REALIZATION isn't Virtual
----------------------------------
(REALIZATION)[met1] String: Hello | Integer : 2
(REALIZATION)New code for met2. Par1: 'Hello'
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
